### ISSUE TITLE
[Bug] Cannot set Neo Backup as the default SMS app

### ISSUE THREAD

**Issue by tatl-im** [NeoApplications/Neo-Backup#858]
**Description**
When trying to restore SMS/MMS JSON special backup, I get an error saying that it's not the default SMS app. However, when I try to change my default SMS app, Neo Backup is not in that list.

I've uninstalled the app, reinstalled version 8.0.0 (the first with SMS backup I believe) and tried every single version from 8.0.0 through 8.3.6.

**Steps To Reproduce**
1. Open Settings, Apps, Default Apps, SMS app
2. Neo Backup is missing

* I also noticed that trying to make a new backup of SMS/MMS also fails--see screenshot

**Expected behavior**
Neo backup should be in the list of SMS apps.

**Screenshots**

First screenshot is the Settings app, default SMS app.
Second is the error when I try to restore SMS.
Third is the error when I try to backup SMS.

![Screenshot_20240410-162441_Permission controller](https://github.com/NeoApplications/Neo-Backup/assets/3693309/0589fd0a-585c-468b-8636-da92f9979f6f)
![Screenshot_20240410-162459_Neo Backup](https://github.com/NeoApplications/Neo-Backup/assets/3693309/60ac839e-1
![Screenshot_20240410-163720_Neo Backup](https://github.com/NeoApplications/Neo-Backup/assets/3693309/3d6817a7-af53-4c77-8ab1-adaa3530b8fd)
9a3-43a2-b45a-24d0cb08ceb0)

Only if there's no clear logs, add screenshots to help explain your problem.

**System Information(please complete the following information):**

- Device: Oneplus 9 Pro
- Android Version: 14
- ROM: crDroid  
- App's Version: 8.0.0 - 8.3.6

-----------------------------

**Additional Notes**

* If you're on the latest Test version(from the TG-Channel or Matrix-Room) please report there directly, not here.

* If you're on the latest github version:
  
  * Please read the [FAQ](https://github.com/NeoApplications/Neo-Backup/blob/master/FAQ.md) first.
  * Please read all the open Issues. If one does match yours, leave a comment there if there's some extra information you can provide.

* If you're not on the latest github version. Update, then make sure that the bug is still there. If it's go further with reporting.

Thanks for reporting.

**Comment by Rorschach1010** [comment]
Same here on CRDroid 10.4. Did did work before an earlier CRDroid A14 versions.

**Comment by keldian** [comment]
Any workaround to restore messages back from the json?

**Comment by Rorschach1010** [comment]
None that I know of. I tried 3 of the well-know opensource SMS apps (Fossify, QUIK and Deku) as some of these apps are able to export and import your sms but none was able to parse and import neos json backup.

This started with Googles QPR2 update as it did work on Android 14 before that update. The strange thing is, none of the SMS apps does have this problem. All are able to be set as the default SMS app, it's just with Neo Backup so Google changed something with their API for SMS permissions on non-SMS apps?

I hope this issue will be resolved (it might even just work again with googles next security patches?). For me personally, the wifi restore issue is way more important tbh.

**Comment by tatl-im** [comment]
> Same—backup on Android 12 did work but can't restore on 14. Any workaround to restore the messages from the json?

I downgraded the app. I don't remember what version, but I just kept rolling it further and further back until it worked.

**Comment by keldian** [comment]
Do you remember going far enough back that the app is called OAndBackupX again? I spent about 30 mins trying older versions but found none that can be set as the SMS app.

**Comment by elea11** [comment]
> This started with Googles QPR2 update as it did work on Android 14 before that update. The strange thing is, none of the SMS apps does have this problem. All are able to be set as the default SMS app, it's just with Neo Backup so Google changed something with their API for SMS permissions on non-SMS apps?
> 

I agree and also assume it is a new background check. Same for me on Google's Android 15 beta (Pixel 7). I guess it is some "security feature" as
I was newly getting the advice to uninstall NB even though I disabled both Play Protect toggles.![Screenshot_20240516-091758.png](https://github.com/NeoApplications/Neo-Backup/assets/61006322/372efad6-1fbe-4239-aec6-4bb8f3d19448)

I will try and get a log to maybe identify what calls are running after a fresh NB installation.

**Comment by tatl-im** [comment]
> Do you remember going far enough back that the app is called OAndBackupX again? I spent about 30 mins trying older versions but found none that can be set as the SMS app.

I think so. I found an "OAndBackupX" folder on my sdcard.



> > This started with Googles QPR2 update as it did work on Android 14 before that update. The strange thing is, none of the SMS apps does have this problem. All are able to be set as the default SMS app, it's just with Neo Backup so Google changed something with their API for SMS permissions on non-SMS apps?
> 
> I agree and also assume it is a new background check. Same for me on Google's Android 15 beta (Pixel 7). I guess it is some "security feature" as I was newly getting the advice to uninstall NB even though I disabled both Play Protect toggles.

I don't even have Google Play Services installed. I should add that to the OP.

**Comment by dl200010** [comment]
For the backup, I suggest checking the permissions in the settings app for Neo Backup. It should work just fine without being able to set to default.

For the restore, it is a current known bug. Google changed something in Android that caused Neo Backup to not show for default. I have fixed this problem. So, in the next release it should show up in SMS default app list.

https://github.com/NeoApplications/Neo-Backup/pull/918
